nodeclient/core_models.go

```diff
@@ -93,113 +108,82 @@ type (
 	BlockMetadataResponse struct {
-		// The hex encoded ID of the block.
+		// BlockID The hex encoded block ID of the block.
 		BlockID string `json:"blockId"`
-		// The hex encoded IDs of the parent block references.
-		Parents []string `json:"parents"`
-		// Whether the block is solid.
-		Solid bool `json:"isSolid"`
-		// The milestone index that references this block.
-		ReferencedByMilestoneIndex iotago.SlotIndex `json:"referencedByMilestoneIndex,omitempty"`
-		// If this block represents a milestone this is the milestone index
-		MilestoneIndex iotago.SlotIndex `json:"milestoneIndex,omitempty"`
-		// The ledger inclusion state of the transaction payload.
-		LedgerInclusionState string `json:"ledgerInclusionState,omitempty"`
-		// Whether the block should be promoted.
-		ShouldPromote *bool `json:"shouldPromote,omitempty"`
-		// Whether the block should be reattached.
-		ShouldReattach *bool `json:"shouldReattach,omitempty"`
-		// The reason why this block is marked as conflicting.
-		ConflictReason uint8 `json:"conflictReason,omitempty"`
-		// If this block is referenced by a milestone this returns the index of that block inside the milestone by whiteflag ordering.
-		WhiteFlagIndex *uint32 `json:"whiteFlagIndex,omitempty"`
-	}
-
-	// ChildrenResponse defines the response of a GET children REST API call.
-	ChildrenResponse struct {
-		// The hex encoded block ID of the block.
-		BlockID string `json:"blockId"`
-		// The maximum count of results that are returned by the node.
-		MaxResults uint32 `json:"maxResults"`
-		// The actual count of results that are returned.
-		Count uint32 `json:"count"`
-		// The hex encoded IDs of the children of this block.
-		Children []string `json:"children"`
+		// StrongParents are the strong parents of the block.
+		StrongParents []string `json:"strongParents"`
+		// WeakParents are the weak parents of the block.
+		WeakParents []string `json:"weakParents"`
+		// ShallowLikeParents are the shallow like parents of the block.
+		ShallowLikeParents []string `json:"shallowLikeParents"`
+		// BlockState might be pending, confirmed, finalized.
+		BlockState string `json:"blockState"`
+		// TxState might be pending, conflicting, confirmed, finalized, rejected.
+		TxState string `json:"txState,omitempty"`
+		// BlockStateReason if applicable indicates the error that occurred during the block processing.
+		BlockStateReason string `json:"blockStateReason,omitempty"`
+		// TxStateReason if applicable indicates the error that occurred during the transaction processing.
+		TxStateReason string `json:"txStateReason,omitempty"`
+		// ReissuePayload whether the block should be reissued.
+		ReissuePayload *bool `json:"reissuePayload,omitempty"`
 	}
 
 	// OutputMetadataResponse defines the response of a GET outputs metadata REST API call.
 	OutputMetadataResponse struct {
-		// The hex encoded ID of the block.
+		// BlockID is the block ID that contains the output.
```

Comment:
What happens if there are multiple included attachments of the same transactions?

Reply:
We returned the earliest block get confirmed.